app/helpers/GoogleTakeOutAutoImport.js

```diff
@@ -2,108 +2,141 @@
  * Checks the download folder, unzips and imports all data from Google TakeOut
  */
 import { subscribe, unzip } from 'react-native-zip-archive';
+import { mergeJSONWithLocalData } from '../helpers/GoogleData';
+import { Platform } from 'react-native';
+import RNFS from 'react-native-fs';
 
-import { MergeJSONWithLocalData } from '../helpers/GoogleData';
+export class NoRecentLocationsError extends Error {}
+export class InvalidFileExtensionError extends Error {}
 
-// require the module
-let RNFS = require('react-native-fs');
-
-// unzipping progress component.
+const ZIP_EXT_CHECK_REGEX = /\.zip$/;
 let progress;
+const MONTHS = [
+  'JANUARY',
+  'FEBRUARY',
+  'MARCH',
+  'APRIL',
+  'MAY',
+  'JUNE',
+  'JULY',
+  'AUGUST',
+  'SEPTEMBER',
+  'OCTOBER',
+  'NOVEMBER',
+  'DECEMBER',
+];
+/**
+ * Safe paths is interested in locations for latest a couple of weeks.
+ * Date for latest 2 months should be sufficient to cover all cases,
+ * especially the case when we are in the early days of the current month.
+ * @returns {string[]} - array of files for latest 2 months from google takeout archive
+ */
+export function getFilenamesForLatest2Months(rootPath) {
+  const now = new Date();
+  const previousMonth = new Date();
+  previousMonth.setMonth(now.getMonth() - 1);
```

Comment:
This totally works. I had to check manually 👍 . Can you add a test that checks this year wrapping case  (i.e its correct in Jan), so that any mods to this code are covered?

Reply:
good catch, thanks. I'll update the logic to rather use MomentJS and cover it in unit test

Reply:
We now use dayJS, which is an immutable momentJS clone